src/components/PopoverButton/Popover.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Manager, Reference, Popper } from 'react-popper';
+import styled from 'react-emotion';
+
+const modifiers = {
+  offset: { enabled: true, offset: '0,10' },
+  flip: { enabled: true }
+};
+
+class Popover extends Component {
+  static propTypes = {
+    /**
+     * function rendering the popover
+     */
+    renderPopover: PropTypes.func.isRequired,
+    /**
+     * function rendering the reference (button or something clickable)
+     */
+    renderReference: PropTypes.func.isRequired,
+    /**
+     * placement of the popover relative to the button
+     */
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
+    /**
+     * alignment of the popover relative to the button
+     */
+    align: PropTypes.oneOf(['start', 'end', 'center'])
+  };
+
+  static defaultProps = {
+    placement: 'bottom',
+    align: 'start'
+  };
+
+  state = { isOpen: false };
+
+  componentDidMount() {
+    document.addEventListener('click', this.onDocumentClick, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onDocumentClick, false);
+  }
+
+  onDocumentClick = ({ target }) => {
+    if (this.popoverRef && this.popoverRef.contains(target)) {
+      return;
+    }
+    if (this.buttonRef && this.buttonRef.contains(target)) {
+      return;
+    }
+
+    this.handleOutsideClick();
+  };
+
+  buttonRef = null;
+  popoverRef = null;
+
+  receiveButtonRef = ref => {
```

Comment:
I don't have an opinion on this but am curious, why not to just inline these in the JSX instead.

Reply:
Had eslint rule against it in my previous Co and .... autopilot :) Also, just tried changing it and definitely don't like how it makes the JSX less readable with introducing two extra lines there.

Generally prefer JSX short and shallow, since it tells you the most about how the component works.